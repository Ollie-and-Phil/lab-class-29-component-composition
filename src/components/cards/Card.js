import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, image }) => {
  return (
    <>
      {title && <header><h3>{title}</h3></header>}
      <figure>
        <img src={image} />
      </figure>
    </>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
};

export default Card;
