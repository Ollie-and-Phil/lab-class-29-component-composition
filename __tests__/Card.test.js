import React from 'react';
import { shallow } from 'enzyme';
import Card from '../src/components/cards/Card';

describe('Card component', () => {
  it('renders Card', () => {
    const wrapper = shallow(<Card name="Phil" image="Phil.jpg" alt="Phil" />);
    expect(wrapper).toMatchSnapshot();
  });
});
