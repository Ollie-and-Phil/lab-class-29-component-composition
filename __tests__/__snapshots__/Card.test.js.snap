// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card 1`] = `
<Fragment>
  <header>
    <h3>
      Phil
    </h3>
  </header>
  <figure>
    <img
      alt="Phil"
      src="Phil.jpg"
    />
  </figure>
</Fragment>
`;
